lib/hex/netrc.ex

```diff
@@ -0,0 +1,15 @@
+defmodule Hex.Netrc do
+  @moduledoc false
+
+  alias Hex.Netrc.Cache
+
+  def lookup(host) when is_binary(host) do
+    case Cache.parse() do
```

Comment:
nitpick: what do you think about calling it a more common for caches `fetch`?

```suggestion
    case Cache.fetch() do
```

btw, instead of `fetch()` should we have `fetch(host)`, that is we look up the machines in the agent process and not in the caller? The benefit is less copying though it definitely doesn't matter much.

Reply:
That's a good idea -- `Cache.fetch` (or `Cache.query`?) sounds like a much more appropriate name! ❤️ 

As for where the lookup is done: in my first iteration, I did indeed have the lookup in the agent process itself. I decided to move it to the caller since I felt that this makes the purpose of the `Hex.Netrc.Cache` module more 'crisp': it _only_ acts as a cache for the parser, it provides no functionality on top of that.

I don't mind changing this though, if you insist on avoiding the copying -- what do you think? 🤔

Reply:
Yeah, even thought it is a micro optimisation (since .netrc files tend to be rather short) I feel like it is more natural to fetch only what you need from the cache, as opposed to fetching everything and then filtering it out in the caller.